### ISSUE TITLE
ipfs add -r directory on Windows uses backslash

### ISSUE THREAD

**Issue by Quiark** [ipfs/go-ipfs#1676]
```
> ipfs add -r overlog.git
QmVUbPiL92L6tp4vVcYQfkB9kmUm2a5QmWKsBWHkzB3NFo 456    overlog.git\COMMIT_EDITMSG
QmVn8wX5YruwN8kd82Qk7eea5b7vw8VLvu2mJc7PCqyyiT 103    overlog.git\FETCH_HEAD
QmWeKwYTKwBwVd7AKioXTmtpLFxTk3MBBk2ef2JtwCccAi 31     overlog.git\HEAD
QmcbSU3g2N5m6uFk8NpXL86BpJo5rmnxirU3ftaSFsadqq 49     overlog.git\ORIG_HEAD
QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn 4      overlog.git\branches/
```

I suppose all folders in ipfs should be named using /

This behaviour is a problem because for example git-remote-ipfs cannot find `overlog.git/refs` but `overlog.git\refs` exists.

**Comment by Quiark** [comment]
digging into the code now...

**Comment by Quiark** [comment]
Oh neat, it includes my commit. It doesn't work though. `ipfs ls $HASH` does list files with normal slash but they aren't actually accessible.

**Comment by whyrusleeping** [comment]
Where do the backslashes come from? are you on windows?

**Comment by Quiark** [comment]
Yes, Windows, that's the point :)

**Comment by jbenet** [comment]
things need to be using `filepath`, not `path` when touching the OS i think?

**Comment by whyrusleeping** [comment]
can be closed now that #1933 is merged.
